tests/visualization_tests/test_pareto_front.py

```diff
@@ -14,439 +14,265 @@
 
-    with pytest.raises(ValueError, match=error_message):
-        plot_pareto_front(
-            study=study,
-            target_names=[],
-            include_dominated_trials=include_dominated_trials,
-            targets=targets,
-        )
+@pytest.mark.parametrize("include_dominated_trials", [False, True])
+@pytest.mark.parametrize("axis_order", [None, [0, 1], [1, 0]])
+@pytest.mark.parametrize("targets", [None, lambda t: (t.values[0], t.values[1])])
+@pytest.mark.parametrize("target_names", [None, ["Foo", "Bar"]])
+def test_get_pareto_front_info_constrained(
+    include_dominated_trials: bool,
+    axis_order: Optional[List[int]],
+    targets: Optional[Callable[[FrozenTrial], Sequence[float]]],
+    target_names: Optional[List[str]],
+) -> None:
+    if axis_order is not None and targets is not None:
+        pytest.skip("skip using both axis_order and targets")
 
-    with pytest.raises(ValueError, match=error_message):
-        plot_pareto_front(
-            study=study,
-            target_names=["Foo"],
-            include_dominated_trials=include_dominated_trials,
-            targets=targets,
-        )
+    study = create_study_2d()
+    trials = study.get_trials(deepcopy=False)
 
-    with pytest.raises(ValueError, match=error_message):
-        plot_pareto_front(
-            study=study,
-            target_names=["Foo", "Bar", "Baz"],
-            include_dominated_trials=include_dominated_trials,
-            axis_order=axis_order,
-            targets=targets,
-        )
+    # (x, y) = (1, 0) is infeasible; others are feasible.
+    def constraints_func(t: FrozenTrial) -> Sequence[float]:
+        return [1.0] if t.params["x"] == 1 and t.params["y"] == 0 else [-1.0]
 
-    target_names = ["Foo", "Bar"]
-    figure = plot_pareto_front(
+    info = _get_pareto_front_info(
         study=study,
-        target_names=target_names,
         include_dominated_trials=include_dominated_trials,
         axis_order=axis_order,
-        constraints_func=constraints_func,
         targets=targets,
+        target_names=target_names,
+        constraints_func=constraints_func,
+    )
+
+    assert info == _ParetoFrontInfo(
+        n_targets=2,
+        target_names=target_names or ["Objective 0", "Objective 1"],
+        best_trials_with_values=[(trials[1], [1, 1]), (trials[2], [0, 2])],
+        non_best_trials_with_values=[(trials[0], [1, 2])] if include_dominated_trials else [],
+        infeasible_trials_with_values=[(trials[3], [1, 0])],
+        axis_order=axis_order or [0, 1],
+        include_dominated_trials=include_dominated_trials,
+        has_constraints_func=True,
     )
-    assert figure.layout.xaxis.title.text == target_names[actual_axis_order[0]]
-    assert figure.layout.yaxis.title.text == target_names[actual_axis_order[1]]
 
 
 @pytest.mark.parametrize("include_dominated_trials", [False, True])
-@pytest.mark.parametrize("use_constraints_func", [False, True])
```

Comment:
That's because the logic of constraints are the same for 3d and 2d, and it would not be simple to parametrize them.

Reply:
I personally think we should cover all the combinations because what to test should not depend too much on the implementation logic, but let's leave its discussion as follow-up work.